pkg/ingester/ingester_v2.go

```diff
@@ -216,6 +242,37 @@ func (u *userTSDB) setLastUpdate(t time.Time) {
 	u.lastUpdate.Store(t.Unix())
 }
 
+func (u *userTSDB) canIdleClose(idleTimeout time.Duration) (bool, error) {
```

Comment:
I would just return `error`, so that when we call it we can log the specific reason why we can't.

Reply:
I considered that, but "not idle" didn't seem like a good error message :)